js/files_test.ts

```diff
@@ -29,3 +29,84 @@ test(async function filesToAsyncIterator() {
 
   assertEqual(totalSize, 12);
 });
+
+testPerm({ write: true }, async function createFile() {
+  const tempDir = await deno.makeTempDir();
+  const filename = tempDir + "/test.txt";
+  let f = await deno.open(filename, deno.OpenMode.Write);
+  let fileInfo = deno.statSync(filename);
+  assert(fileInfo.isFile());
+  assert(fileInfo.len === 0);
+  const enc = new TextEncoder();
+  const data = enc.encode("Hello");
+  await f.write(data);
+  fileInfo = deno.statSync(filename);
+  assert(fileInfo.len === 5);
+  f.close();
+
+  // TODO: test different modes
+  await deno.removeAll(tempDir);
+});
+
+testPerm({ write: true }, async function openModeWrite() {
+  const tempDir = deno.makeTempDirSync();
+  const encoder = new TextEncoder();
+  const filename = tempDir + "hello.txt";
+  const data = encoder.encode("Hello world!\n");
+
+  let file = await deno.open(filename, deno.OpenMode.Write);
+  // assert file was created
+  let fileInfo = deno.statSync(filename);
+  assert(fileInfo.isFile());
+  assertEqual(fileInfo.len, 0);
+  // write some data
+  await file.write(data);
+  fileInfo = deno.statSync(filename);
+  assertEqual(fileInfo.len, 13);
+  // assert we can't read from file
+  let thrown = false;
+  try {
+    const buf = new Uint8Array(20);
+    await file.read(buf);
+  } catch (e) {
+    thrown = true;
+  } finally {
+    assert(thrown, "OpenMode.Write shouldn't allow to read file");
+  }
+  file.close();
+  // assert that existing file is truncated on open
+  file = await deno.open(filename, deno.OpenMode.Write);
+  file.close();
+  const fileSize = deno.statSync(filename).len;
+  assertEqual(fileSize, 0);
+  await deno.removeAll(tempDir);
+});
+
+testPerm({ write: true }, async function openModeWriteRead() {
+  const tempDir = deno.makeTempDirSync();
+  const encoder = new TextEncoder();
+  const filename = tempDir + "hello.txt";
+  const data = encoder.encode("Hello world!\n");
+
+  let file = await deno.open(filename, deno.OpenMode.WriteRead);
+  // assert file was created
+  let fileInfo = deno.statSync(filename);
+  assert(fileInfo.isFile());
+  assertEqual(fileInfo.len, 0);
+  // write some data
+  await file.write(data);
+  fileInfo = deno.statSync(filename);
+  assertEqual(fileInfo.len, 13);
+
+  // TODO: this test is not working, I expect because
+  //  file handle points to the end of file, but ATM
+  //  deno has no seek implementation on Rust side
+  // assert file can be read
+  // const buf = new Uint8Array(20);
+  // const result = await file.read(buf);
+  // console.log(result.eof, result.nread);
+  // assertEqual(result.nread, 13);
+  // file.close();
+
```

Comment:
@ry I hit a roadblock here. Could you take a look at text in TODO and tell if that statement is true? If so, then we need to implement `seek` to fully test this PR.

Reply:
Yes, I think you're correct. It's fine to leave this as a TODO for now.